.github/workflows/rust.yml

```diff
@@ -70,6 +70,18 @@ jobs:
     - uses: dtolnay/rust-toolchain@1.60.0
     - run: cargo build
 
+  build-nostd:
+    name: Build on no_std target (thumbv7em-none-eabi)
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: stable
+          targets: thumbv7em-none-eabi
+      - run: cargo build --target thumbv7em-none-eabi --release --no-default-features --features alloc
+      - run: cargo build --target thumbv7em-none-eabi --release --no-default-features --features "alloc digest rand_core batch zeroize pkcs8"
```

Comment:
Should we consider `cargo hack` here?

Reply:
Yeah we would have to cache it ? is there a github action that includes it or did dtolnay's toolchain action give it ?

Reply:
Hmm it seems to burn a lot of CI minutes as it requires `--each-feature ` ?

We can do either that or just keep two sets of matrix 1) for alloc 2) without alloc

Reply:
Ok cargo hack seems to work - here is fail as expected with serde:
https://github.com/dalek-cryptography/ed25519-dalek/actions/runs/4395018249/jobs/7696481185
Will put the serde in there now and it should succeed here - can re-base and then take the serde exception off there

Reply:
Ahh it doesn't group the features correctly.. we need the features grouped so they activate each other.

Reply:
Ahh need to put `--skip` for the ones which would be pointless alone - 

The feature-cascade should enable each other but do we want to check lone features ? 

Probably best to not complicate it me thinks and keep just simple exclude.

Reply:
What features are pointless alone?

Reply:
Hmm good question :sweat_smile: yeah none I think - I have to do little mental mindmap about the feat matrix's